### ISSUE TITLE
Concern about the validation of training strategy

### ISSUE THREAD

**Issue by lyf1212** [Natsu6767/DCGAN-PyTorch#2]
Are you sure you make a right training of you model? In my view, the loss jitters to much.
Can you spend a little time to fix it?

**Comment by rafa-br34** [comment]
I have the same problem here but I wrote my code based on the pytorch example and it still doesn't work.
From my limited understanding, the discriminator loss should always be quite low but never 0 (otherwise the generator can't learn) and will slowly increase as the generator improves.
For example, here's the graph on the original pytorch tutorial
![sphx_glr_dcgan_faces_tutorial_002](https://github.com/user-attachments/assets/0b9896b9-658f-46ed-aaea-6f07a4c42f1f)
looks perfect right?
welp, here's what I get
![image](https://github.com/user-attachments/assets/956b0c7a-73a4-439e-a446-3eddb62f3f1e)
I don't know what to do.....

**Comment by lyf1212** [comment]
> I have the same problem here but I wrote my code based on the pytorch example and it still doesn't work. From my limited understanding, the discriminator loss should always be quite low but never 0 (otherwise the generator can't learn) and will slowly increase as the generator improves. For example, here's the graph on the original pytorch tutorial ![sphx_glr_dcgan_faces_tutorial_002](https://private-user-images.githubusercontent.com/111343658/361181761-0b9896b9-658f-46ed-aaea-6f07a4c42f1f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MDA1OTgsIm5iZiI6MTcyNTYwMDI5OCwicGF0aCI6Ii8xMTEzNDM2NTgvMzYxMTgxNzYxLTBiOTg5NmI5LTY1OGYtNDZlZC1hYWVhLTZmMDdhNGM0MmYxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNlQwNTI0NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZjhjNWM4NTBkY2Y0YjdiMzRjZmIyYjU5YTBjOTdlNzI5NjQ1YzRmZjYxYzY5NzA3ZGZkMWEwMThjNzcxMGZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9PpToXPr3CxHEsguCuup1D2dh1jKSIdQVmcUaT4olD8) looks perfect right? welp, here's what I get ![image](https://private-user-images.githubusercontent.com/111343658/361185396-956b0c7a-73a4-439e-a446-3eddb62f3f1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MDA1OTgsIm5iZiI6MTcyNTYwMDI5OCwicGF0aCI6Ii8xMTEzNDM2NTgvMzYxMTg1Mzk2LTk1NmIwYzdhLTczYTQtNDM5ZS1hNDQ2LTNlZGRiNjJmM2YxZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNlQwNTI0NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZmMzNjQ4MDM2YmJkYTU0MWRmNDNjYjkzYjRhYmExNjg5YTg3MTc0NDFlNTY5YzJkNmU1MDIzZDBhZDhmZTBhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.C9FNcaqbJffdIu7qETYwWlS0jFRbkh-XMRRQKRaxQIg) I don't know what to do.....

As it's known well, GAN training is confusing and painful sometimes. If you still want to try, adjust learning rate, optimizer hyperparameters or other settings, and print necessary logs during training, not only loss values, but also image outputs. But in my opinion, DCGAN indeed made a remarkable progress, it;s meaningless to retrain it with limited computational resource. FYI.

**Comment by rafa-br34** [comment]
> As it's known well, GAN training is confusing and painful sometimes. If you still want to try, adjust learning rate, optimizer hyperparameters or other settings, and print necessary logs during training, not only loss values, but also image outputs. But in my opinion, DCGAN indeed made a remarkable progress, it;s meaningless to retrain it with limited computational resource. FYI.

I agree, the only reason why I'm trying to train a DCGAN is that (in my opinion) it seems to be the best way to get hands-on experience with adversarial loss, most modern architectures use this type of loss (high-fidelity VAEs for instance), but I can't deny that I was much more successful at training a barely functional SD model than any GANs so far lol.
